staging/src/k8s.io/kubectl/pkg/cmd/debug/debug.go

```diff
@@ -467,12 +504,90 @@ func (o *DebugOptions) debugByEphemeralContainer(ctx context.Context, pod *corev
 	return result, debugContainer.Name, nil
 }
 
+// applyCustomProfile applies given partial container json file on to the profile
+// incorporated debug pod.
+func (o *DebugOptions) applyCustomProfile(debugPod *corev1.Pod, containerName string) error {
```

Comment:
The inputs to these functions are the outputs of `o.generatePod....`. Does it make sense to move the applies into those functions? Feel free to tell me I'm crazy or there's a good reason. Something wasn't sitting right with me while trying to grok what was going on here.

Reply:
I think, this is a good idea. Moved custom apply into `generate...` functions.